File: app/src/main/java/co/com/cesarnorena/pokedex/database/PokedexDatabase.java
package co.com.cesarnorena.pokedex.database;

import com.raizlabs.android.dbflow.annotation.Database;

/**
 * Created by Cesar Norena on 29/02/2016.
 */
@Database(name = PokedexDatabase.NAME, version = PokedexDatabase.VERSION)
public class PokedexDatabase {

    public static final String NAME = "pokedexapp";
    public static final int VERSION = 1;
}

File: app/src/main/kotlin/co/cesarnorena/pokedex/data/local/PokedexDatabase.kt
package co.cesarnorena.pokedex.data.local

import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context

@Database(
        entities = [PokemonEntity::class],
        version = 1, exportSchema = false
)
abstract class PokedexDatabase : RoomDatabase() {

    abstract fun pokemonDao(): PokemonDao

    companion object {
        fun getPokemonDao(context: Context): PokemonDao {
            return Room.databaseBuilder(context, PokedexDatabase::class.java, PokemonEntity.NAME)
                    .build().pokemonDao()
        }
    }
}